### PR TITLE
Add LM head and masked fill_mask for bert-burn.

### LINKED ISSUE #30: RoBERTa weights do not have encoders with norm_first: true
Through tracing `transformers` running the same model on a fill-mask task, I was able to determine that the execution of `transformers` and `bert-burn` diverge at the point where normalization happens.

Furthermore, using the lm_head weights for roberta-base and attaching a LM head model, I was able to verify that bert-burn's results are correct with `norm_first: false`, but entirely wrong for `norm_first: true`.

I'd be happy to provide a pull request, but I'm not sure whether other BERT models _do_ use `norm_first: true`. I'm very new to machine learning and am not familiar with this family of models.

### PR DESCRIPTION
**Pull request by seurimas**
I'm not sure if this matches the structure desired for this crate, but I had figured out masked token inference for a personal project and it seemed like a waste not to try and contribute. Please let me know what sort of changes might be needed to get this merged, if this is something worth merging.

Additionally, this fixed #30.

### CODE REVIEW

File: bert-burn/src/fill_mask.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#[derive(Debug, Clone)]
+pub struct FillMaskResult {
+    pub mask_idx: usize,
+    pub top_k: Vec<(f32, String)>,
+}
+
+pub fn fill_mask<B: Backend>(
+    model: &BertMaskedLM<B>,
+    model_config: &BertModelConfig,
+    tokenizer: &Arc<BertTokenizer>,
```
Comment (nathanielsimard) [inline bert-burn/src/fill_mask.rs:47]: You can remove the Arc when only taking the reference and never cloning it.

File: bert-burn/src/fill_mask.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        let masks = find_masks(&input_tokens, MASK_TOKEN_ID);
+        for mask in masks {
+            let logits = output
+                .clone()
+                .slice([i..i + 1, mask..(mask + 1), 0..d_model])
+                .squeeze::<2>(0)
+                .squeeze(0);
+            // Find the top k tokens with the highest probabilities
+            let probs = data_to_vec_f32(&softmax(logits, 0).to_data());
+            let top_k = top_k(5, probs);
```
Comment (nathanielsimard) [inline bert-burn/src/fill_mask.rs:75]: You could sort the softmax instead of doing it on the CPU: `sort_with_indices` would work fine I think (where the indice is the token and the value is the prob).

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
I like it! I have some minor comments, but it should be ready to merge after they are fixed.

**Review by nathanielsimard — APPROVED** [review]
